private_sharing/models.py

```diff
@@ -74,6 +88,36 @@ def badge_upload_path(instance, filename):
     return 'direct-sharing/badges/{0}/{1}'.format(instance.id, filename)
 
 
+def project_membership_visible(user, project):
+    """
+    Determine if the user's membership in a project is visible or not.
+    """
+    project_id = int(id_label_to_project(project).id)
+    user_id = user_to_id(user)
+
+    if project_id is not None:
+        if DataRequestProjectMember.objects.filter(member_id=user_id).exists():
+            project = DataRequestProjectMember.objects.get(member_id=user_id,
+                                                           project_id=project_id)
+            return bool(project.visible)
+
+    return False
+
+
+def toggle_membership_visibility(user, project, state):
+    """
+    Change the state of whether a member's data sharing is publicly visible or not.
+    """
+    project_id = int(project)
+    user_id = user_to_id(user)
+    state = bool(strtobool(state))
+    project = DataRequestProjectMember.objects.get(member_id=user_id,
```

Comment:
Member IDs and user IDs are different. This function seems to be trying to accept both User and Member objects. Assuming what's actually generated with the user_to_id function is a User object ID, it isn't guaranteed to map to a Member ID (indeed, they're all off-by-one in our current database).

And... usually it's trivial to look one up from the other, `user.member` and `member.user`, so I think the function could just accept one (and then the variable name would be more accurate here) and in contexts where there's currently a member passed, pass the `.user` for that thing instead. (This also removes the complexity of the `user_to_id` code – which is confusing anyway, because "user" is being used to refer to things that aren't User objects, guarenteed to confuse I'm afraid! :confounded: